js/settingsview.js

```diff
@@ -16,40 +16,78 @@
 		+ '	{{/unless}}'
 		+ '</div>';
 
+	/**
+	 * @class OC.Settings.TwoFactorU2F.View
+	 */
 	var View = Backbone.View.extend({
+
+		/**
+		 * @type {function|undefined}
+		 */
 		_template: undefined,
+
+		/**
+		 * @type {boolean}
+		 */
 		_enabled: undefined,
+
+		/**
+		 * @type {boolean}
+		 */
 		_loading: false,
-		template: function(data) {
+
+		/**
+		 * @param {object} data
+		 * @returns {string}
+		 */
+		template: function (data) {
 			if (!this._template) {
 				this._template = Handlebars.compile(TEMPLATE);
 			}
 			return this._template(data);
 		},
+
 		events: {
 			'change #u2f-enabled': '_onToggleEnabled'
 		},
-		initialize: function() {
+
+		/**
+		 * @returns {undefined}
+		 */
+		initialize: function () {
 			this._load();
 		},
-		render: function() {
+
+		/**
+		 * @returns {undefined}
+		 */
+		render: function () {
 			this.$el.html(this.template({
 				enabled: this._enabled,
 				loading: this._loading
 			}));
 		},
-		_load: function() {
+
+		/**
+		 * @returns {undefined}
+		 */
+		_load: function () {
 			var url = OC.generateUrl('/apps/twofactor_u2f/settings/state');
-			$.ajax(url, {
+			Promise.resolve($.ajax(url, {
```

Comment:
I think jquery implements enough of Promises/A to allow you to chain it with `then`, but I'm not sure about using it with other promise implementations